packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1859,19 +1869,19 @@ export class ContainerRuntime extends EventEmitter
             // Note: Chunking will increase content beyond maxOpSize because we JSON'ing JSON payload -
             // there will be a lot of escape characters that can make it up to 2x bigger!
             // This is Ok, because DeltaManager.shouldSplit() will have 2 * maxMessageSize limit
-            if (serializedContent.length <= maxOpSize) {
+            if (!serializedContent || serializedContent.length <= maxOpSize) {
                 clientSequenceNumber = this.submitRuntimeMessage(
                     type,
                     content,
                     this._flushMode === FlushMode.Manual,
-                    batchBegin ? { batch: true } : undefined);
+                    batchBegin ? { batch: true } : opMetadata);
             } else {
                 clientSequenceNumber = this.submitChunkedMessage(type, serializedContent, maxOpSize);
             }
         }
 
         // Let the PendingStateManager know that a message was submitted.
-        this.pendingStateManager.onSubmitMessage(type, clientSequenceNumber, content, localOpMetadata);
+        this.pendingStateManager.onSubmitMessage(type, clientSequenceNumber, content, opMetadata ?? localOpMetadata);
```

Comment:
This seems fishy. I think pendingStateManager needs to know about both - opMetadata & localOpMetadata.
The former is needed if we need to resubmit an op, right?